data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleFactoryImpl.java

```diff
@@ -202,19 +186,46 @@ public AbstractVerticle get(final DataPlaneContract.Resource resource, final Dat
     return getConsumerVerticle(deliveryOrder, initializer, new HashSet<>(resource.getTopicsList()));
   }
 
-  static ResponseHandler getNoopResponseHandlerOrDefault(final DataPlaneContract.Egress egress,
-                                                         final Supplier<ResponseHandler> defaultSupplier) {
-    if (egress.hasDiscardReply()) {
+  static ResponseHandler getResponseHandler(final DataPlaneContract.Egress egress,
+                                            final Supplier<ResponseHandler> kafkaSupplier,
+                                            final Supplier<ResponseHandler> httpSupplier) {
+    if(egress.hasReplyUrl()){
+      return httpSupplier.get();
+    } else if(egress.hasReplyToOriginalTopic()){
+      return kafkaSupplier.get();
+    } else if (egress.hasDiscardReply()) {
       return new NoopResponseHandler();
     }
-    return defaultSupplier.get();
+    //TODO: log
+    return kafkaSupplier.get();
```

Comment:
I had to carry over the logic because so many things failed when I changed the logic to return `Noop` when nothing was set.

I think we should do one of these:
- Check for requiredness of the `replyStrategy` and reject when it is not set
- If not, at least default to `Noop` and not to Kafka

I don't want to do this in this PR though.

If you guys agree, I will create an issue about that.

Reply:
> Check for requiredness of the replyStrategy and reject when it is not set
> If not, at least default to Noop and not to Kafka

I'd do the second one since we don't have a way of telling the control plane that the reply strategy wasn't set properly.

In general, yes, controllers must set the reply field (we can safety checks to the common logic to update the CM in the controller)

Reply:
I see this as an implementation problem and after giving some thought, I would prefer #1.

Created a ticket here: https://github.com/knative-sandbox/eventing-kafka-broker/issues/1725 where we can discuss more